controller/src/test/java/org/jboss/as/controller/operation/global/EnhancedSyntaxTestCase.java

```diff
@@ -728,83 +712,91 @@ public void testRuntimeListAttributeWrite() throws Exception {
         op = createOperation("write-attribute", TEST_ADDRESS);
         op.get("name").set(RUNTIME_LIST_ATTRIBUTE.getName() + "[-1]");
         op.get("value").set("list-value2");
-        executeCheckNoFailure(op);
-
-        list = StringListAttributeDefinition.unwrapValue(ExpressionResolver.TEST_RESOLVER, runtimeListAttributeValue);
-        Assert.assertEquals(2, list.size());
-        Assert.assertEquals("list-value", list.get(0));
-        Assert.assertEquals("list-value2", list.get(1));
-
-        // :write-attribute(name=runtime-list-attribute[0])
-        op = createOperation("write-attribute", TEST_ADDRESS);
-        op.get("name").set(RUNTIME_LIST_ATTRIBUTE.getName() + "[0]");
-        op.get("value").set("list-value3");
-        executeCheckNoFailure(op);
-
-        list = StringListAttributeDefinition.unwrapValue(ExpressionResolver.TEST_RESOLVER, runtimeListAttributeValue);
-        Assert.assertEquals(2, list.size());
-        Assert.assertEquals("list-value3", list.get(0));
-        Assert.assertEquals("list-value2", list.get(1));
+        executeCheckForFailure(op);
     }
 
     @Test
     public void testCompositeListWrite() throws Exception {
-        performCompositeListWrite(LIST_ATTRIBUTE.getName());
-        performCompositeListWrite(RUNTIME_LIST_ATTRIBUTE.getName());
```

Comment:
I don't get the point of this removal yet. The difference between `LIST_ATTRIBUTE` vs `RUNTIME_LIST_ATTRIBUTE` is the runtime attribute condition on the latter one. Both attributes have an attribute definition, so my initial understanding is the test should pass even if we are now throwing an exception here https://github.com/wildfly/wildfly-core/pull/4451/files#diff-761bfd4f7ad6fa2a357742d89610c21973e1732d39f3225a8f5274f9c26165f2R97.

@petrberan any clue about why should the test fails using the second one?

In general, at first glance maybe it could be better to adapt the test cases to any new situation instead of removing cases. So, if it is expected to fail, then assert it fails. However, as I said, it is a bit suspicious that it fails with the runtime one and passes with the other.

Reply:
First of all, thank you for your advice of using assert fail, that was a really bad pattern on my side. I'll make sure to use it properly next time. 

As for the issue why one runtime test fails and non-runtime doesn't, it all comes down to WriteAttributeHandler. On server side, only "attribute-list" exists and both test are using attributes like "attribute-list[0]". In WriteAttributeHandler, both attributes pass useEnhancedSyntax check (there's [0] in attribute name) and thus attributes without suffix [0] ("attribute-list") are created (line 81). 

However, in case of runtime test, this new attribute is not used at all and the executed operation has the old one "attribute-list[0]" parameter instead (line 121). This attribute does not exists and, exception is thrown and test fails. 

In case of non-runtime test, the operation executes with "attribute-list" parameter, this attribute exists and operation does not fail (line 116).

To be honest, I don't really understand why this is happening. I've spent a lot of time debugging the whole process trying to understand it, but ultimately considered this an intended behaviour.

I've talked about this issue with my superior rstancel today and we should be looking into this - presumably tomorrow. But if you have any idea why this is happening please let me know.

Reply:
Turned out I've read your message wrong. I believe I didn't remove any test cases, I had to split some tests into two, since runtime one would fail and the non-runtime one didn't, but both cases should be covered. This time I also check if error message is correct.

We've looked into this issue with Radovan and found out that in WriteAttributeHandler on line 181 there's created a clone of runtime operation, which is resolved without EnhancedSyntax (186), but then the original operation is executed with EnhancedSyntax (197), causing runtime tests to fail. 

Given this issue, we consider this as an indented behaviour.

Reply:
Yes, I believe we are understanding the same goal for this issue. I think I may have grasped it from a different perspective and my solution is way too nested and thus incorrect. So instead of failing at AbstractWriteAttributeHandler in case of no definition for an attribute, it could fail already in WriteAttributeHandler, perhaps just after doExecuteInternal at 171 in case you've described, if attributeAccess has no attribute definition